java/arcs/core/storage/DirectStoreMuxer.kt

```diff
@@ -120,6 +126,9 @@ class DirectStoreMuxer<Data : CrdtData, Op : CrdtOperation, T>(
     suspend fun store(id: String) = storeMutex.withLock {
         stores.getOrPut(id) {
             setupStore(id)
+        }.also {
+            val toClose = closeList.getAndSet(emptyList())
+            toClose.forEach { closeStore(it) }
```

Comment:
If the only time we call `closeStore` is when this method is called or when `clearStoresCache`, it means we need to guarantee that `clearStoresCache` is always done at the end of this class's lifecycle, right? It might be good to add a test to the ReferenceModeStore tests to check that this is the case.

Reply:
I think this might be covered. The cleanup strategy for referenceModeStore right now is to clear the backing store when the receiveQueue is empty and there are no callbacks. We have tests for that behavior in `ReferenceModeStoreTest`. Is this what you mean?